### PR TITLE
feature(tiger): Протягивание текстовых ячеек

### PR DESCRIPTION
**Pull request by dontuse**
https://jira.railsc.ru/browse/GOODS-486

### CODE REVIEW

File: src/Table/reducers/selected.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+};
+
+export default (state = initialState, action) => {
+  switch (action.type) {
+    case types.TABLE_EDITOR_CELL_SELECT_START:
+      return {
+        ...state,
+        ids: [action.payload.id],
+        idTo: [],
+        name: action.payload.name,
```
Comment (apress-hound) [inline src/Table/reducers/selected.js]: Missing semicolon.<br>Unrecoverable syntax error. (29% scanned).

File: src/Table/reducers/selected.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  isDragging: false,
+};
+
+export default (state = initialState, action) => {
+  switch (action.type) {
+    case types.TABLE_EDITOR_CELL_SELECT_START:
+      return {
+        ...state,
+        ids: [action.payload.id],
+        idTo: [],
```
Comment (apress-hound) [inline src/Table/reducers/selected.js]: Label 'idTo' on [ statement.

File: src/Table/reducers/selected.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  isSelected: false,
+  isDragging: false,
+};
+
+export default (state = initialState, action) => {
+  switch (action.type) {
+    case types.TABLE_EDITOR_CELL_SELECT_START:
+      return {
+        ...state,
+        ids: [action.payload.id],
```
Comment (apress-hound) [inline src/Table/reducers/selected.js]: Missing semicolon.<br>Expected 'case' and instead saw ':'.<br>Expected '}' to match '{' from line 10 and instead saw ':'.<br>Expected an identifier and instead saw ','.

File: src/Table/reducers/selected.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  idTo: [],
+  isSelected: false,
+  isDragging: false,
+};
+
+export default (state = initialState, action) => {
+  switch (action.type) {
+    case types.TABLE_EDITOR_CELL_SELECT_START:
+      return {
+        ...state,
```
Comment (apress-hound) [inline src/Table/reducers/selected.js]: Expected '}' to match '{' from line 13 and instead saw '...'.<br>Missing semicolon.

File: src/Table/reducers/selected.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
+import * as types from '../actions';
+
+const initialState = {
+  ids: [],
+  idTo: [],
+  isSelected: false,
+  isDragging: false,
+};
+
+export default (state = initialState, action) => {
```
Comment (apress-hound) [inline src/Table/reducers/selected.js]: Regular parameters should not come after default parameters.

File: src/Table/reducers/selected.js (hunk 1 above)
Comment (apress-hound) [inline src/Table/reducers/selected.js]: Missing semicolon.<br>Unrecoverable syntax error. (29% scanned).

File: src/Table/reducers/selected.js (hunk 2 above)
Comment (apress-hound) [inline src/Table/reducers/selected.js]: Label 'idTo' on [ statement.

File: src/Table/reducers/selected.js (hunk 3 above)
Comment (apress-hound) [inline src/Table/reducers/selected.js]: Missing semicolon.<br>Expected 'case' and instead saw ':'.<br>Expected '}' to match '{' from line 10 and instead saw ':'.<br>Expected an identifier and instead saw ','.

File: src/Table/reducers/selected.js (hunk 4 above)
Comment (apress-hound) [inline src/Table/reducers/selected.js]: Expected '}' to match '{' from line 13 and instead saw '...'.<br>Missing semicolon.

File: src/Table/reducers/selected.js (hunk 5 above)
Comment (apress-hound) [inline src/Table/reducers/selected.js]: Regular parameters should not come after default parameters.

File: src/Table/reducers/selected.js (hunk 1 above)
Comment (apress-hound) [inline src/Table/reducers/selected.js]: Missing semicolon.<br>Unrecoverable syntax error. (29% scanned).

File: src/Table/reducers/selected.js (hunk 2 above)
Comment (apress-hound) [inline src/Table/reducers/selected.js]: Label 'idTo' on [ statement.

File: src/Table/reducers/selected.js (hunk 3 above)
Comment (apress-hound) [inline src/Table/reducers/selected.js]: Missing semicolon.<br>Expected 'case' and instead saw ':'.<br>Expected '}' to match '{' from line 10 and instead saw ':'.<br>Expected an identifier and instead saw ','.

File: src/Table/reducers/selected.js (hunk 4 above)
Comment (apress-hound) [inline src/Table/reducers/selected.js]: Expected '}' to match '{' from line 13 and instead saw '...'.<br>Missing semicolon.

File: src/Table/reducers/selected.js (hunk 5 above)
Comment (apress-hound) [inline src/Table/reducers/selected.js]: Regular parameters should not come after default parameters.

File: src/Table/reducerRow.js [hunk 6]
```diff
@@ -25,6 +26,40 @@ export default function rows(state = [], action) {
         return row;
       });
 
+    case TABLE_EDITOR_CELL_END_DRAG: {
+      let CellsFrom = [];
```
Comment (Kupov) [inline src/Table/reducerRow.js]: сellsFrom это же не компонент

File: src/Table/reducers/selected.js [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+        ...state,
+        isSelected: false,
+        isDragging: true,
+      };
+
+    case types.TABLE_EDITOR_CELL_SELECT_ADD:
+      return {...state, ids: state.ids.concat(action.payload.id)};
+
+    case types.TABLE_EDITOR_CELL_SELECT_ADD_TO:
+      return {...state, idTo: state.idTo.concat(action.payload.id)};
```
Comment (Kupov) [inline src/Table/reducers/selected.js]: [...state.idTo, action.payload.id]

Comment (Kupov) [inline src/Table/reducers/selected.js]: массивы так же как объекты можно мерджить

Comment (Kupov) [inline src/Table/reducers/selected.js]: ну и в остальных местах тоже

File: src/Table/reducerRow.js (hunk 6 above)
Comment (dontuse) [inline src/Table/reducerRow.js]: ну это ячейки из которых копируют, поэтому так называется или ты про что ?

Comment (dontuse) [inline src/Table/reducerRow.js]: аа, с большой буквы, вижу да да ....

File: src/Table/reducers/selected.js (hunk 1 above)
Comment (apress-hound) [inline src/Table/reducers/selected.js]: Missing semicolon.<br>Unrecoverable syntax error. (29% scanned).

File: src/Table/reducers/selected.js (hunk 2 above)
Comment (apress-hound) [inline src/Table/reducers/selected.js]: Label 'idTo' on [ statement.

File: src/Table/reducers/selected.js (hunk 3 above)
Comment (apress-hound) [inline src/Table/reducers/selected.js]: Missing semicolon.<br>Expected 'case' and instead saw ':'.<br>Expected '}' to match '{' from line 10 and instead saw ':'.<br>Expected an identifier and instead saw ','.

File: src/Table/reducers/selected.js (hunk 4 above)
Comment (apress-hound) [inline src/Table/reducers/selected.js]: Expected '}' to match '{' from line 13 and instead saw '...'.<br>Missing semicolon.

File: src/Table/reducers/selected.js (hunk 5 above)
Comment (apress-hound) [inline src/Table/reducers/selected.js]: Regular parameters should not come after default parameters.

File: src/Table/reducers/selected.js (hunk 1 above)
Comment (apress-hound) [inline src/Table/reducers/selected.js]: Missing semicolon.<br>Unrecoverable syntax error. (29% scanned).

File: src/Table/reducers/selected.js (hunk 2 above)
Comment (apress-hound) [inline src/Table/reducers/selected.js]: Label 'idTo' on [ statement.

File: src/Table/reducers/selected.js (hunk 3 above)
Comment (apress-hound) [inline src/Table/reducers/selected.js]: Missing semicolon.<br>Expected 'case' and instead saw ':'.<br>Expected '}' to match '{' from line 10 and instead saw ':'.<br>Expected an identifier and instead saw ','.

File: src/Table/reducers/selected.js (hunk 4 above)
Comment (apress-hound) [inline src/Table/reducers/selected.js]: Expected '}' to match '{' from line 13 and instead saw '...'.<br>Missing semicolon.

File: src/Table/reducers/selected.js (hunk 5 above)
Comment (apress-hound) [inline src/Table/reducers/selected.js]: Regular parameters should not come after default parameters.

**Pull request merged**